File: BackEnd/src/domain/student.ts
import { GradeId, UserId, StudentId } from './ids';

export enum ParentRelation {
  FATHER = 'Father',
  MOTHER = 'Mother',
  GUARDIAN = 'Guardian',
}

export class Student {
  studentId: StudentId;
  fullEnglishName: string;
  fullArabicName: string;
  nationalId: string;
  parentId?: UserId;
  parentLinkCode: string;
  gradeId: GradeId;
  schoolName?: string;
  parentRelation?: ParentRelation;
  isActive: boolean;
}

File: Frontend/GP/Models/Student.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class Student : BaseEntity
    {
        [Key]
        public int StudentId { get; set; }
        public string FullEnglishName { get; set; }
        public string FullArabicName { get; set; }
        public int NationalId { get; set; }
        public National National { get; set; }
        public int ParentId { get; set; }
        public Parent Parent { get; set; }
        public string ParentLinkCode { get; set; }
        public int GradeId { get; set; }
        public Grade Grade { get; set; }    
        public string? SchoolName { get; set; }
        public string? ParentRelation { get; set; }
        public bool IsActive { get; set; }
    }
}
